### PR TITLE
Fix/infinite scroll

### PR DESCRIPTION
**Pull request by omareloui**
## Description

Fixes the search, and infinite scrolling, and rely on getting the data from the backend.

This PR branched out from #377.

## Type of Change

- [x] Bug fix
- [x] New feature
- [ ] Test case
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [x] I have added necessary documentation (if applicable)

## 4- steps to test ?

Please provide a brief summary of the steps required to test the changes in this PR.

## 5- results ?

Please provide a brief summary of the results after testing the changes in this PR.

## 7- screenshots ?

Please provide screenshots of the results after testing the changes in this PR.

### CODE REVIEW

File: src/containers/feedOfClaim/index.tsx [hunk 1]
```diff
@@ -83,84 +85,99 @@ const SourceLink = ({ claim, searchTerm }: { claim: LocalClaim; searchTerm: stri
...
+
+const FeedClaim: React.FC<IHomeProps> = () => {
+  const [claims, setClaims] = useState<ImportedClaim[]>([])
+  const claimsRef = useRef<ImportedClaim[]>([])
+
   const location = useLocation()
-  const [isAuth, setIsAuth] = useState(false)
-  const [isLoading, setIsLoading] = useState(false)
+  // const [isAuth, setIsAuth] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
```
Comment (eimaam) [inline src/containers/feedOfClaim/index.tsx:106]:
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(true)
```

File: src/containers/feedOfClaim/index.tsx [hunk 2]
```diff
@@ -83,84 +85,99 @@ const SourceLink = ({ claim, searchTerm }: { claim: LocalClaim; searchTerm: stri
...
-  const [showScrollButton, setShowScrollButton] = useState(false)
+  const [searchTerm, setSearchTerm] = useState(getSearchFromParams() || '')
   const navigate = useNavigate()
   const theme = useTheme()
   const isMediumScreen = useMediaQuery(theme.breakpoints.down('md'))
-  const isAuthenticated = checkAuth()
 
-  useEffect(() => {
+  const [showScrollButton, setShowScrollButton] = useState(false)
+  // const isAuthenticated = checkAuth()
```
Comment (eimaam) [inline src/containers/feedOfClaim/index.tsx:115]: Is this needed? We can get rid of it if not needed and if it is, lets add a //**TODO** comment

Comment (eimaam) [inline src/containers/feedOfClaim/index.tsx:115]: Is this needed? We can get rid of it if not needed and if it is, lets add a //**TODO** comment

Comment (omareloui) [reply src/containers/feedOfClaim/index.tsx:115]: It was there from the PR I've branched from, I'm not sure why it was there

File: src/containers/feedOfClaim/index.tsx (hunk 1 above)
Comment (omareloui) [reply src/containers/feedOfClaim/index.tsx:106]: Redundant. It's inferred from the parameter.

File: src/containers/feedOfClaim/index.tsx (hunk 2 above)
Comment (omareloui) [reply src/containers/feedOfClaim/index.tsx]: Removed it

Comment (omareloui) [reply src/containers/feedOfClaim/index.tsx]: Removed it

**Review by AhmedAbdelmenam — APPROVED** [review]

**Pull request merged**